src/lib-components/DateFilter.vue

```diff
@@ -0,0 +1,607 @@
+}
+// Close datepicker
+function closeDatepicker() {
+  datepicker.value?.closeMenu()
+}
+defineExpose({
+  openDatepicker,
+  closeDatepicker,
+})
+
+// COMPUTED VALUES
+// Determine if the window size is mobile for conditional rendering
+const isMobile = computed(() => globalStore.winWidth <= 750)
+// Format the selected date(s) into consistent object
+const formattedDateSelection = computed(() => {
+  // range selected
+  if (date.value && 'length' in date.value) {
+    return {
+      startDate: date.value[0],
+      endDate: date.value[1],
+    }
+  }
+  // single date selected
+  else if (date.value) {
+    return {
+      startDate: date.value,
+      endDate: null,
+    }
+  }
+  // nothing selected
+  return {
+    startDate: null,
+    endDate: null,
+  }
+})
+
+// ASYNC COMPONENTS
+const ButtonLink = defineAsyncComponent(() =>
+  import('@/lib-components/ButtonLink.vue'))
+
+// Watch date and emit to parent component
+watch(date, async (newDate, oldDate) => {
+  if (newDate !== oldDate) {
+    // change seperator for single or range select
+    if ((newDate && 'length' in newDate) && newDate[1] === null) {
+      textConfig.value = {
+        rangeSeparator: ' ',
+      }
+    }
+    else {
+      textConfig.value = {
+        rangeSeparator: ' — ',
+      }
+    }
+    // eslint-disable-next-line vue/custom-event-name-casing
+    emit('input-selected', formattedDateSelection.value)
+  }
+})
+// Watch window size and update windowSize ref
+/* window.addEventListener('resize', () => {
+  windowSize.value = window.innerWidth
+}) */
+</script>
+
+<template>
+  <div class="date-filter-container">
+    <VueDatePicker
+      ref="datepicker" v-model="date" :config="vue3datepickerConfig" :range="!isMobile" :week-start="0"
+      month-name-format="long" :enable-time-picker="false" :auto-position="false" :auto-apply="true"
+      :text-input="textConfig" no-today :inline="hideInput" class="date-filter"
+      :class="[{ 'is-selecting': isSelecting }]" :placeholder="isMobile ? 'Select a date' : 'All upcoming'"
+      @internal-model-change="handleInternalSelection" @range-start="clearTodayBtn" @open="toggleArrow"
+      @closed="toggleArrow"
+    >
+      <template #input-icon>
+        <SvgIconFTVACalender />
+        <span class="toggle-triangle-icon" :class="[{ 'is-open': isOpen }]">
+          <SvgIconFTVADropTriangle />
+        </span>
+      </template>
+
+      <template #clear-icon="{ clear }">
+        <SvgIconClose @click="clear" />
+      </template>
+
+      <template
+        #month-year="{
+          month,
+          year,
+          months,
+          handleMonthYearChange,
+        }"
+      >
+        <div class="custom-header">
+          <div class="custom-month-year-component">
+            {{ months[month].text }} {{ year }}
+          </div>
+          <div class="custom-nav-buttons">
+            <button class="nav-arrow-button" @click="() => { clearTodayBtn(); handleMonthYearChange(false) }">
```

Comment:
move the click logic to script setup method/function

Reply:
I've revised this line to 
`<button class="nav-arrow-button" @click="clearTodayBtn(); handleMonthYearChange(false)">`

this way I can avoid the inline function, but still use the handleMonthYearChange() method that is exposed by the vue3datepicker #month-year slot. Using it outside of the slot seems to prevent the mothod from being found.

Hopefully this is ok